src/Components/Users/UserProfile.tsx

```diff
@@ -182,14 +181,39 @@ export default function UserProfile() {
                 Notification.Success({
                     msg: "Details updated successfully"
                 });
-                window.location.reload();
+                setDetails({ ...details, first_name: states.form.firstName, last_name: states.form.lastName, age: states.form.age, gender: genderTypes.filter(el => { return el.id === Number(states.form.gender) })[0].text, email: states.form.email, phone_number: states.form.phoneNumber });
+                setShowEdit(false);
             }
         }
     };
 
-    return (
+    const checkForNewBuildVersion = async () => {
+
+        let [isLatestBundle, newVersion] = await checkIfLatestBundle();
+
+        if (!isLatestBundle) {
+            setUpdateBtnText("updating...");
+            localStorage.setItem('build_meta_version', newVersion);
+
+            if ('caches' in window) {
+                // Service worker cache should be cleared with caches.delete()
+                caches.keys().then(names => {
+                    for (const name of names) {
+                        caches.delete(name);
+                    }
 
-        <div >
+                    window.location.reload(true);
+                });
+            }
+        } else {
+            setUpdateBtnText("You already have the latest version!")
+
+            setTimeout(() => setUpdateBtnText("Update"), 1000);
```

Comment:
Codacy found an issue: [The function setTimeout can be unsafe](https://app.codacy.com/gh/coronasafe/care_fe/pullRequest?prid=5838477)